03-task-manager/starter/app.js

```diff
@@ -1 +1,39 @@
-console.log('Task Manager App')
+const express = require("express");
+const app = express();
+const tasks = require("./routes/tasks");
+const connectDB = require("./db/connect");
+require("dotenv").config();
+const port = 3000;
+
+// A middleware built into ExpressJS is express.json(). If we dont use this,
+// we won't have that data in req.body.
+app.use(express.json());
+
+
+// routes
+app.get("/hello", (req, res) => {
+    res.send("Task Manager App")
+});
+
+app.use("/api/v1/tasks", tasks);
+
+
+// app.get("/api/v1/tasks")  - get all the tasks
+// app.post("/api/v1/tasks")  - create a new task
+// app.get("/api/v1/tasks:id")  - get single task
+// app.patch("/api/v1/tasks:id")  - update task
+// app.delete("/api/v1/tasks:id")  - delete task
+
+// So that only connection to DB is successful, then we can spin off the server
+const start = async() => {
+    try {
+        await connectDB(process.env.MONGO_URI)
+        app.listen(port, console.log(`Server is listening on port ${port}...`))
+
+    } catch (error) {
+        console.log(error)
+    }
+};
+// to access the dotenv anywhere, use process.env.<packageName>
```

Comment:
Just to clarify - `dotenv` is a library that adds the variables in the provided `.env` file to the built-in Node `process.env` objection. When you are calling `process.env.VARIABLE` you are just accessing an object with properties on it. (dotenv is only used to modify that object).